.github/workflows/gem-build-and-install.yml

```diff
@@ -0,0 +1,95 @@
+name: Test gem build and installation
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  cancel-in-progress: true
+
+on:
+  pull_request:
+
+jobs:
+  native-package:
+    name: Compile native gem
+    runs-on: ubuntu-latest
+    strategy:
+      # fail-fast: true
+      matrix:
+        platform:
+          - x86_64-linux
+          # - arm-linux # needs https://github.com/matsadler/magnus/issues/17 ", "aarch64-linux", "x86_64-darwin", "arm64-darwin", "x64-mingw32", "x64-mingw-ucrt"
+          # - "aarch64-linux"
+
+          - "x86_64-darwin"
+          # - "arm64-darwin" # github actions does not support this
+
+          # - "x64-mingw-ucrt"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: oxidize-rb/actions/setup-ruby-and-rust@main
+        with:
+          ruby-version: "3.1"
+          rubygems: latest
+          bundler-cache: false
+          cargo-cache: true
+          cargo-vendor: true
+
+      - uses: oxidize-rb/cross-gem-action@main
+        with:
+          platform: ${{ matrix.platform }}
+          env: |
+            RUBY_CC_VERSION=3.1.0
+
+      - name: "Test gem build"
+        run: |
+          ./script/test-gem-build gems ${{matrix.platform}}
+
+      - uses: actions/upload-artifact@v3
```

Comment:
cross-gem-action does perform an upload of the artifact, so this may not be necessary?